smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -21,6 +20,24 @@ contract MixinPurchase is
 {
   event RenewKeyPurchase(address indexed owner, uint newExpiration);
 
+  event GasRefunded(address indexed receiver, uint refundedAmount, address tokenAddress);
```

Comment:
A specific event is emitted so things can be tracked, but we can also just track transfer event to save gas

Reply:
Let's keep the event for now.